utils/warnings.ml

```diff
@@ -220,8 +220,14 @@ let is_active x = (!current).active.(number x);;
 let is_error x = (!current).error.(number x);;
 
 let parse_opt error active flags s =
-  let set i = flags.(i) <- true in
-  let clear i = flags.(i) <- false in
+  let set i =
+    let i = if i = 25 then 8 else i in
+    flags.(i) <- true
+  in
+  let clear i =
+    let i = if i = 25 then 8 else i in
+    flags.(i) <- false
```

Comment:
It is indeed quite bad.

My preference would go to keeping the change to `set`, revert the change to `clear` and not emit the warning.
If that sounds OK to you I'll do that then.

(And considering that there is an actual code change, perhaps I should add an entry to Changes actually?)